lib/util/http/host-rules.ts

```diff
@@ -157,6 +155,10 @@ export function applyHostRule<GotOptions extends HostRulesGotOptions>(
     options.abortIgnoreStatusCodes = hostRule.abortIgnoreStatusCodes;
   }
 
+  if (hostRule.enabled === false) {
+    options.enabled = false;
+  }
+
```

Comment:
I think this can be moved to start of function and an early return added. if it's disabled all other code is irrelevant and can be skipped.